### PR TITLE
Reorganize tests to make them easier to find

### PR DESCRIPTION
**Pull request by aspfohl**
Moves tests into folders that roughly mimic the source code structure
Renames test files where it seemed appropriate, to match source code structure
Remove adding "scripts" to path several times - instead do it on test setup
Some adjustments to path-based logic that broke down after moving files

### CODE REVIEW

**Review by dakinggg — APPROVED** [review]
LGTM whenever it passes CI. Thank you!

File: tests/a_scripts/__init__.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Copyright 2022 MosaicML LLM Foundry authors
```
Comment (irenedea) [inline tests/a_scripts/__init__.py:1]: why is it `a_scripts`?

**Review by irenedea — APPROVED** [review]

File: tests/a_scripts/inference/test_convert_composer_to_hf.py [hunk 2]
```diff
@@ -399,7 +391,7 @@ def test_huggingface_conversion_callback(model: str, tmp_path: pathlib.Path,
                                          expected_normal_checkpoints: int):
     delete_transformers_cache()
 
-    dist.initialize_dist(get_device('gpu'))
+    # dist.initialize_dist(get_device('gpu'))
```
[the remark this answers is not in the document]
Comment (aspfohl) [reply tests/a_scripts/inference/test_convert_composer_to_hf.py:394]: Removing it didn't change anything.. I may just add it back 😛

[the remark this answers is not in the document]
Comment (dakinggg) [reply tests/a_scripts/inference/test_convert_composer_to_hf.py:394]: nah, it shouldn't be there. just remove please

File: tests/a_scripts/__init__.py (hunk 1 above)
Comment (aspfohl) [reply tests/a_scripts/__init__.py:1]: Bit of a hack right now - mysterious gpu failure if script tests run last https://github.com/mosaicml/llm-foundry/actions/runs/7055479688/job/19206776145

Comment (aspfohl) [reply tests/a_scripts/__init__.py:1]: Everything looks green.. going to add a comment and come back to this later 🙂

**Pull request merged**